src/main/java/delete_task_use_case/DeleteTaskDataAccess.java

```diff
@@ -14,11 +14,11 @@
 
 public class DeleteTaskDataAccess extends WeekDataAccess implements DeleteTaskDsGateway {
     @Override
-    public void save(DeleteTaskDataAccessInput dataAccessInput) {
-        DataAccessDay day = super.getDays().get(dataAccessInput.getDayIndex());
+    public void save(DeleteTaskDsInputData dataAccessInput) {
+        DataAccessDay day = days.get(dataAccessInput.getDayIndex());
         HashMap<String, DataAccessTask> tasks = day.getTasks();
-        tasks.remove(dataAccessInput.getTaskTitle());
+        DataAccessTask deleted = tasks.remove(dataAccessInput.getTaskTitle());
 
-        super.save();
+        this.save();
```

Comment:
Hi Ricky. You should indeed be calling super.save(), and not this.save(), which I believe invokes the overriden implementation of save() in your DataAccess class.

Reply:
Reverted change, as suggested.